packages/wallet/src/W3mFrameProvider.ts

```diff
@@ -89,8 +90,20 @@ export class W3mFrameProvider {
   }
 
   // -- Extended Methods ------------------------------------------------
+  public getLoginEmailUsed() {
+    return Boolean(W3mFrameStorage.get(W3mFrameConstants.EMAIL_LOGIN_USED_KEY))
+  }
+
   public async connectEmail(payload: W3mFrameTypes.Requests['AppConnectEmailRequest']) {
     await this.w3mFrame.frameLoadPromise
+    const lastEmailLoginTime = W3mFrameStorage.get(W3mFrameConstants.LAST_EMAIL_LOGIN_TIME)
+    if (lastEmailLoginTime) {
+      const difference = W3mFrameHelpers.getTimeDifferenceMs(Number(lastEmailLoginTime))
+      if (difference < 30_000) {
+        const cooldownSec = Math.ceil((30_000 - difference) / 1000)
+        throw new Error(`Please try again after ${cooldownSec} seconds`)
```

Comment:
Wouldn't it be better to show an error to the user instead of throwing an Error here? It'll break the app as I know, right?